internal/provider/cluster_resource_test.go

```diff
@@ -618,13 +618,19 @@ func TestIntegrationDedicatedClusterResource(t *testing.T) {
 	finalizedCluster := upgradingCluster
 	finalizedCluster.UpgradeStatus = client.CLUSTERUPGRADESTATUSTYPE_FINALIZED
 
+	scaledCluster := finalizedCluster
+	scaledCluster.Config.Dedicated = &client.DedicatedHardwareConfig{}
+	*scaledCluster.Config.Dedicated = *finalizedCluster.Config.Dedicated
```

Comment:
Something about these three lines feels funky, but I think I'm just having a mental block. If you get a moment, I'd really appreciate a small comment explaining what this does :)

Reply:
Removed this whole section in favor of updating `cluster` directly. But in case you're curious, when copying `finalizedCluster`, it copied `Config` by reference, so I had to create a new config and copy the value over to avoid mutating `finalizedCluster`.